test/extended/olm/olmv1-webhook-support.go

```diff
@@ -0,0 +1,299 @@
+		o.Expect(spec).To(o.Equal(map[string]interface{}{
+			"valid":  true,
+			"mutate": true,
+		}))
+	})
+
+	g.It("should have a working conversion webhook", func(ctx g.SpecContext) {
+		file := filepath.Join(webhookTestBaseDir, "conversion-webhook-test.yaml")
+		g.By("creating a valid webhook test resource")
+		cleanup, err := apply(oc, file)
+		g.DeferCleanup(cleanup)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		cfg, err := clientcmd.BuildConfigFromFlags("", exutil.KubeConfigPath())
+		o.Expect(err).ToNot(o.HaveOccurred())
+		c, err := dynamic.NewForConfig(cfg)
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		v2WebhookTestClient := c.Resource(schema.GroupVersionResource{
+			Group:    "webhook.operators.coreos.io",
+			Version:  "v2",
+			Resource: "webhooktests",
+		})
+
+		g.By("getting the created resource in v2 schema")
+		obj, err := v2WebhookTestClient.Namespace(testNamespace).Get(ctx, conversionWebhookTestResourceName, metav1.GetOptions{})
+		o.Expect(err).ToNot(o.HaveOccurred())
+		o.Expect(obj).ToNot(o.BeNil())
+
+		g.By("validating the resource spec")
+		spec := obj.Object["spec"].(map[string]interface{})
+		o.Expect(spec).To(o.Equal(map[string]interface{}{
+			"conversion": map[string]interface{}{
+				"valid":  true,
+				"mutate": true,
+			},
+		}))
+	})
+
+	g.It("should still work through openshift-service-ca certificate rotation", func(ctx g.SpecContext) {
+		file := filepath.Join(webhookTestBaseDir, "mutating-webhook-test.yaml")
+
+		// delete tls secret
+		g.By("deleting the openshift-service-ca signing-key secret")
+		err := oc.AsAdmin().WithoutNamespace().Run("delete").Args("secret", openshiftServiceCASigningKeySecretName, "--namespace", openshiftServiceCANamespace).Execute()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// even with the secret deleted the webhook should still be responsive
+		g.By("checking webhook is responsive through cert rotation")
+		o.Eventually(func(g o.Gomega) {
+			cleanup, err := apply(oc, file)
+			g.Expect(err).ToNot(o.HaveOccurred())
+			cleanup()
+		}).Should(o.Succeed(), "15s", "3s")
+	})
+
+	g.It("should still work if tls secret is deleted", func(ctx g.SpecContext) {
+		file := filepath.Join(webhookTestBaseDir, "mutating-webhook-test.yaml")
+
+		g.By("ensuring secret exists")
+		o.Eventually(func(g o.Gomega) {
+			err := oc.AsAdmin().WithoutNamespace().Run("get").Args("secret", certificateSecretName, "-n", testNamespace).Execute()
+			o.Expect(err).ToNot(o.HaveOccurred())
+		}).Should(o.Succeed())
+
+		// delete tls secret
+		g.By("checking webhook is responsible through secret recreation")
+		err := oc.AsAdmin().WithoutNamespace().Run("delete").Args("secret", certificateSecretName, "-n", testNamespace).Execute()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		// even with the secret deleted the webhook should still be responsive
+		o.Consistently(func(g o.Gomega) {
+			cleanup, err := apply(oc, file)
+			g.Expect(err).ToNot(o.HaveOccurred())
+			cleanup()
```

Comment:
The fact that this test uses `Consistently` and the other test uses `Eventually` seems asymmetric. I'd either use the same test semantic in both cases or add a comment explaining why they are different.